app/src/main/java/com/readrops/app/more/preferences/PreferencesScreenModel.kt

```diff
@@ -42,15 +57,40 @@ class PreferencesScreenModel(
                         openLinksWith = (list[4] as String) to openLinksWith,
                         timelineItemSize = (list[5] as String) to timelineItemSize,
                         mainFilterPref = (list[6] as String) to mainFilter,
-                        syncAtLaunchPref = (list[7] as Boolean) to synchAtLaunch
+                        syncAtLaunchPref = (list[7] as Boolean) to synchAtLaunch,
+                        useCustomShareIntentTpl = (list[8] as Boolean) to useCustomShareIntentTpl,
+                        customShareIntentTpl = (list[9] as String) to customShareIntentTpl,
+                        exampleItem = if (database.itemDao().count() > 0) {
+                            database.itemDao().selectFirst()
+                        } else {
+                            Item(
+                                title = context.getString(R.string.example_item_title),
+                                author = context.getString(R.string.example_item_author),
+                                content = context.getString(R.string.example_item_content),
+                                link = "https://example.org"
+                            )
+                        }
                     )
                 }.collect { theme ->
-                    mutableState.update { theme }
+                    mutableState.update { previous ->
+                        (previous as? PreferencesScreenState.Loaded)?.let {
+                            theme.copy(showDialog = previous.showDialog)
+                        } ?: theme
+                    }
```

Comment:
I am not sure it it a good idea to refresh the whole state when on preference changes. This seem to produce unnecessary screen refreshes as well as pulled the rug from under my feet by resetting the `showDialog` state. The dialog just wouldn't display at all until I realised that.

Reply:
Compose is smart enough to know which fields have been mutated. You can verify with the layout inspector tool. You will see that only `ListPreferenceWidget` are recomposed, the others are skipped. This might be due to lambda unstability.